nmostesting/IS07Utils.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2019 Advanced Media Workflow Association
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import TestHelper
+
+from .NMOSUtils import NMOSUtils
+
+
+class IS07Utils(NMOSUtils):
+    def __init__(self, url):
+        NMOSUtils.__init__(self, url)
+
+    def get_sources_states_and_types(self):
+        """Gets a list of the available source objects with state and type on the API"""
+        toReturn = {}
+        sources_url = self.url + "sources/"
+        valid_sources, sources = TestHelper.do_request("GET", sources_url)
+        if valid_sources:
+            for source in sources.json():
+                source_id = source[:-1]
+                toReturn[source_id] = {}
+                for sub_path in ["state", "type"]:
+                    valid_sub, sub = TestHelper.do_request("GET", "{}/{}/{}".format(sources_url, source_id, sub_path))
+                    if not valid_sub or sub.status_code != 200:
+                        pass
```

Comment:
Yup. Well spotted. I think checking for valid is enough.

Reply:
I think the check of `sub.status_code` is a good idea, it just needs to be `continue` instead of `pass`, to move onto the next `sub_path` (or out of that loop).